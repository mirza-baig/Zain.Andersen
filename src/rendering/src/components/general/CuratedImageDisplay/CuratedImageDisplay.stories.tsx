// Global
import { Story, Meta } from '@storybook/react';
// Local
import CuratedImageDisplay, { CuratedImageDisplayProps } from './CuratedImageDisplay';
import defaultData from './CuratedImageDisplay.mock-data';

export default {
  title: 'General/CuratedImageDisplay',
  component: CuratedImageDisplay,
} as Meta;
const Template: Story<CuratedImageDisplayProps> = (props) => <CuratedImageDisplay {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
