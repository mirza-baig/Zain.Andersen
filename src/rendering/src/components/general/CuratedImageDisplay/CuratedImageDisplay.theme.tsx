import { ThemeFile } from 'lib/context/ThemeContext';

export const CuratedImageDisplayTheme: ThemeFile = {
  aw: {
    classes: {
      headline: 'text-theme-text text-s md:text-m font-heavy mb-s',
      bodyCopy: 'text-theme-body text-body mb-s font-regular',
      SingleButton: {
        wrapper: '',
        cta1Classes: '',
      },
    },
  },
  rba: {
    classes: {
      headline: 'text-theme-text text-sm-m md:text-l font-extra-light mb-s md:mb-xxs',
      bodyCopy: 'text-theme-body text-body mb-s md:mb-m font-regular',
      SingleButton: {
        wrapper: 'mb-m md:mb-l',
        cta1Classes: '',
      },
    },
  },
};
