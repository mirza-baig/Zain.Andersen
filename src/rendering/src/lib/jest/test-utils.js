// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
