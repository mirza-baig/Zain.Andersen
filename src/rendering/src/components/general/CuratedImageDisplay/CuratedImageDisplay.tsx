// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { ImageField, Text, withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
import { CuratedImageDisplayTheme } from './CuratedImageDisplay.theme';
import { ComponentProps } from 'lib/types/component-props';
import { getEnum } from 'lib/utils';
import { Headline } from 'src/helpers/Headline';
import { BodyCopy } from 'src/helpers/BodyCopy';
import SingleButton from 'src/helpers/SingleButton/SingleButton';
import { ImagePrimary } from 'src/helpers/Media';
import { SliderWrapper } from 'src/helpers/SliderWrapper';
import { useCurrentScreenType, getBreakpoint } from 'lib/utils/get-screen-type';
import React, { useRef, useState } from 'react';
import ModalWrapper from 'src/helpers/ModalWrapper/ModalWrapper';
import { getPhotoItemProps } from 'src/helpers/PhotoItemWithDetail/PhotoItemWithDetail.Utils';
import PhotoItemWithDetail, {
  PhotoItemWithDetailProps,
} from 'src/helpers/PhotoItemWithDetail/PhotoItemWithDetail';
import { SvgIcon } from 'src/helpers/SvgIcon';
import { Foundation } from 'src/.generated/Foundation.EnterpriseWeb.model';
import { SliderRefType, SliderType } from 'src/helpers/SliderWrapper/SliderWrapper';

export type CuratedImageDisplayProps =
  Feature.EnterpriseWeb.Enterprise.Components.General.CuratedImageDisplay.CuratedImageDisplay & {
    fields?: {
      images: Foundation.EnterpriseWeb.Enterprise.BaseTemplates.BasePhoto[];
    };
  } & ComponentProps;

type displayStyle = 'grid' | 'horizontal-scroll';

const CuratedImageDisplay = (props: CuratedImageDisplayProps) => {
  const [currentSlideIndex, setCurrentSlideIndex] = useState(0);
  const [isLightboxVisible, setIsLightboxVisible] = useState(false);

  const sliderRef = useRef<SliderType>();

  const sliderSettings = {
    initialSlide: props.fields?.images && props.fields?.images.length > 1 ? currentSlideIndex : 0,
    arrows: false,
    dots: false,
  };

  const openModal = (index: number) => {
    setIsLightboxVisible(true);
    setCurrentSlideIndex(index);
  };

  const { themeData } = useTheme(CuratedImageDisplayTheme);
  const { currentScreenWidth } = useCurrentScreenType();

  const DESKTOP_DISPLAY_STYLE = getEnum<displayStyle>(props.fields?.desktopDisplayStyle) || 'grid';
  const MOBILE_DISPLAY_STYLE = getEnum<displayStyle>(props.fields?.mobileDisplayStyle) || 'grid';

  const renderCuratedImages = (): JSX.Element => {
    if (currentScreenWidth < getBreakpoint('md')) {
      // Renderings for mobile devices
      switch (MOBILE_DISPLAY_STYLE) {
        case 'grid':
          return gridStyleImages();
        case 'horizontal-scroll':
          return scrollStyleImage();
      }
    } else {
      // Renderings for tablets and large screen devices
      switch (DESKTOP_DISPLAY_STYLE) {
        case 'grid':
          return gridStyleImages();
        case 'horizontal-scroll':
          return scrollStyleImage();
      }
    }
  };

  const gridStyleImages = (): JSX.Element => {
    return (
      <ul className="grid grid-cols-2 gap-y-m md:grid-cols-12 md:gap-s">
        {props.fields?.images.map(
          (
            ImageItem: Foundation.EnterpriseWeb.Enterprise.BaseTemplates.BasePhoto,
            index: number
          ) => {
            return (
              <li
                tabIndex={0}
                key={index}
                onClick={() => {
                  openModal(index);
                }}
                onKeyDown={(e) => {
                  if (e.key === 'Enter') {
                    openModal(index);
                  }
                }}
                className="group relative col-span-12 cursor-pointer md:col-span-4 lg:col-span-3"
              >
                <ImagePrimary
                  imageLayout={'responsive'}
                  ratio="square"
                  fields={{
                    primaryImageCaption: {
                      value: '',
                    },
                    primaryImage: ImageItem.fields?.thumbnailImage as ImageField,
                    primaryImageMobile: ImageItem.fields?.thumbnailImage as ImageField,
                  }}
                />
                <SvgIcon
                  className="-translate-t-3/4 absolute top-1/2 left-1/2 -translate-x-1/2 rounded-full bg-black bg-opacity-[.65] p-l text-white opacity-0 transition-all ease-linear group-hover:-translate-y-1/2 group-hover:opacity-100"
                  icon="zoom-pinch"
                />
              </li>
            );
          }
        )}
      </ul>
    );
  };

  const scrollStyleImage = (): JSX.Element => {
    const sliderSettings = {
      dots: false,
      arrow: true,
      infinite: false,
      slidesToShow: 4,
      responsive: [
        {
          breakpoint: getBreakpoint('md'),
          settings: {
            slidesToShow: 1,
            dots: true,
          },
        },
        {
          breakpoint: getBreakpoint('lg'),
          settings: {
            slidesToShow: 3,
            dots: false,
          },
        },
      ],
    };

    return (
      <SliderWrapper sliderSettings={sliderSettings}>
        {props.fields?.images.map(
          (
            ImageItem: Foundation.EnterpriseWeb.Enterprise.BaseTemplates.BasePhoto,
            index: number
          ) => {
            return (
              <React.Fragment key={`ImageItem-${index}`}>
                <div
                  tabIndex={0}
                  onClick={() => {
                    openModal(index);
                  }}
                  onKeyDown={(e) => {
                    if (e.key === 'Enter') {
                      openModal(index);
                    }
                  }}
                  className="group relative cursor-pointer md:mr-s"
                >
                  <ImagePrimary
                    imageLayout={'responsive'}
                    ratio="square"
                    fields={{
                      primaryImageCaption: {
                        value: '',
                      },
                      primaryImage: ImageItem.fields?.thumbnailImage as ImageField,
                      primaryImageMobile: ImageItem.fields?.thumbnailImage as ImageField,
                    }}
                  />
                  <SvgIcon
                    className="-translate-t-3/4 absolute top-1/2 left-1/2 -translate-x-1/2 rounded-full bg-black bg-opacity-[.65] p-l text-white opacity-0 transition-all ease-linear group-hover:-translate-y-1/2 group-hover:opacity-100"
                    icon="zoom-pinch"
                  />
                </div>
              </React.Fragment>
            );
          }
        )}
      </SliderWrapper>
    );
  };

  const renderPhotoItem = () => {
    return props.fields?.images?.map(
      (
        result: Foundation.EnterpriseWeb.Enterprise.BaseTemplates.BasePhoto &
          (
            | Feature.EnterpriseWeb.AndersenWindows.Data.Photos.Photo
            | Feature.EnterpriseWeb.RenewalByAndersen.Data.Photos.Photo
          ),
        index: number
      ) => {
        const photoObject = getPhotoItemProps(result, false) as PhotoItemWithDetailProps;

        return <PhotoItemWithDetail key={index} {...photoObject} />;
      }
    );
  };

  return (
    <Component variant="lg" dataComponent="general/curatedimagedisplay" {...props}>
      <div className="col-span-12">
        <Headline {...props} classes={themeData.classes.headline} />
        <BodyCopy {...props} classes={themeData.classes.bodyCopy} />
        <SingleButton {...props} classes={themeData.classes.SingleButton} />
        {renderCuratedImages()}
        {/* Photo gallery */}
        {isLightboxVisible && (
          <ModalWrapper
            isModalOpen={isLightboxVisible}
            size="fluid"
            handleClose={() => setIsLightboxVisible(false)}
          >
            <div className="px-ml pb-ml pt-s">
              {props.fields?.images && props.fields?.images.length > 1 ? (
                <>
                  <SliderWrapper
                    sliderSettings={sliderSettings}
                    sliderRef={sliderRef as SliderRefType}
                  >
                    {renderPhotoItem()}
                  </SliderWrapper>
                  <div className="mt-m flex items-center justify-between text-xxs md:justify-center">
                    <div
                      role="button"
                      className="ml-xxxs flex cursor-pointer items-center text-button font-bold md:mr-xs"
                      onClick={() => sliderRef.current && sliderRef.current.slickPrev()}
                    >
                      <SvgIcon className="mr-xs" icon="arrow-left" />

                      <Text field={props.fields?.previousLabel} tag="span" />
                    </div>
                    <div
                      role="button"
                      className="mr-xxxs flex cursor-pointer items-center text-button font-bold md:ml-xs"
                      onClick={() => {
                        sliderRef.current && sliderRef.current.slickNext();
                      }}
                    >
                      <Text field={props.fields?.nextLabel} tag="span" />
                      <SvgIcon className="ml-xs" icon="arrow-right" />
                    </div>
                  </div>
                </>
              ) : (
                renderPhotoItem()
              )}
            </div>
          </ModalWrapper>
        )}
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<CuratedImageDisplayProps>(CuratedImageDisplay);
