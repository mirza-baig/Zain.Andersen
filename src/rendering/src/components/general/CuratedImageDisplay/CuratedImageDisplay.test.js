// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
import 'lib/jest/match-media-mock';
// Local
import CuratedImageDisplay from './CuratedImageDisplay';
import defaultData from './CuratedImageDisplay.mock-data';

it('renders correctly', async () => {
  const component = snapshot(CuratedImageDisplay, { componentProps: defaultData });
  await hasDataComponent(component, 'general/curatedimagedisplay');
});
