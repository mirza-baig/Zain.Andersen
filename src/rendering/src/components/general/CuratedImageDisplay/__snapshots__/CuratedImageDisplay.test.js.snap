// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/curatedimagedisplay"
  id="id1647234774"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="col-span-12"
      >
        <div
          class="items-top font-sans [&_a:hover]:underline text-theme-text text-s md:text-m font-heavy mb-s"
        >
          <h2>
            Headline text goes here
          </h2>
        </div>
        <div
          class="text-theme-body text-body mb-s font-regular"
        >
          <div
            class="body-copy"
          >
            <div>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
            </div>
          </div>
        </div>
        <div
          class="mb-s flex items-start md:flex-row md:items-center"
        >
          <a
            aria-label="Call to Action (Opens in a new tab)"
            class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
            data-gtm-click=""
            data-gtm-dl-event="cta_click"
            href="/"
            rel="noopener noreferrer"
            role="link"
            tabindex="0"
            target="_blank"
            title=""
          >
            Call to Action
            <span
              class="ml-xxs"
            />
            <span
              class="sr-only"
            >
               (Opens in a new tab)
            </span>
          </a>
        </div>
        <ul
          class="grid grid-cols-2 gap-y-m md:grid-cols-12 md:gap-s"
        >
          <li
            class="group relative col-span-12 cursor-pointer md:col-span-4 lg:col-span-3"
            tabindex="0"
          >
            <div
              class="relative hidden md:block aspect-square   "
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <span
              class="-translate-t-3/4 absolute top-1/2 left-1/2 -translate-x-1/2 rounded-full bg-black bg-opacity-[.65] p-l text-white opacity-0 transition-all ease-linear group-hover:-translate-y-1/2 group-hover:opacity-100"
            />
          </li>
          <li
            class="group relative col-span-12 cursor-pointer md:col-span-4 lg:col-span-3"
            tabindex="0"
          >
            <div
              class="relative hidden md:block aspect-square   "
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <span
              class="-translate-t-3/4 absolute top-1/2 left-1/2 -translate-x-1/2 rounded-full bg-black bg-opacity-[.65] p-l text-white opacity-0 transition-all ease-linear group-hover:-translate-y-1/2 group-hover:opacity-100"
            />
          </li>
          <li
            class="group relative col-span-12 cursor-pointer md:col-span-4 lg:col-span-3"
            tabindex="0"
          >
            <div
              class="relative hidden md:block aspect-square   "
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <span
              class="-translate-t-3/4 absolute top-1/2 left-1/2 -translate-x-1/2 rounded-full bg-black bg-opacity-[.65] p-l text-white opacity-0 transition-all ease-linear group-hover:-translate-y-1/2 group-hover:opacity-100"
            />
          </li>
          <li
            class="group relative col-span-12 cursor-pointer md:col-span-4 lg:col-span-3"
            tabindex="0"
          >
            <div
              class="relative hidden md:block aspect-square   "
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <span
              class="-translate-t-3/4 absolute top-1/2 left-1/2 -translate-x-1/2 rounded-full bg-black bg-opacity-[.65] p-l text-white opacity-0 transition-all ease-linear group-hover:-translate-y-1/2 group-hover:opacity-100"
            />
          </li>
          <li
            class="group relative col-span-12 cursor-pointer md:col-span-4 lg:col-span-3"
            tabindex="0"
          >
            <div
              class="relative hidden md:block aspect-square   "
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <span
              class="-translate-t-3/4 absolute top-1/2 left-1/2 -translate-x-1/2 rounded-full bg-black bg-opacity-[.65] p-l text-white opacity-0 transition-all ease-linear group-hover:-translate-y-1/2 group-hover:opacity-100"
            />
          </li>
          <li
            class="group relative col-span-12 cursor-pointer md:col-span-4 lg:col-span-3"
            tabindex="0"
          >
            <div
              class="relative hidden md:block aspect-square   "
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <span
              class="-translate-t-3/4 absolute top-1/2 left-1/2 -translate-x-1/2 rounded-full bg-black bg-opacity-[.65] p-l text-white opacity-0 transition-all ease-linear group-hover:-translate-y-1/2 group-hover:opacity-100"
            />
          </li>
          <li
            class="group relative col-span-12 cursor-pointer md:col-span-4 lg:col-span-3"
            tabindex="0"
          >
            <div
              class="relative hidden md:block aspect-square   "
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <span
              class="-translate-t-3/4 absolute top-1/2 left-1/2 -translate-x-1/2 rounded-full bg-black bg-opacity-[.65] p-l text-white opacity-0 transition-all ease-linear group-hover:-translate-y-1/2 group-hover:opacity-100"
            />
          </li>
          <li
            class="group relative col-span-12 cursor-pointer md:col-span-4 lg:col-span-3"
            tabindex="0"
          >
            <div
              class="relative hidden md:block aspect-square   "
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <span
              class="-translate-t-3/4 absolute top-1/2 left-1/2 -translate-x-1/2 rounded-full bg-black bg-opacity-[.65] p-l text-white opacity-0 transition-all ease-linear group-hover:-translate-y-1/2 group-hover:opacity-100"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>
`;
